client/cli/src/lib.rs

```diff
@@ -243,12 +247,22 @@ pub fn init_logger(
 	pattern: &str,
 	tracing_receiver: sc_tracing::TracingReceiver,
 	profiling_targets: Option<String>,
+	disable_log_reloading: bool,
 ) -> std::result::Result<(), String> {
-	fn parse_directives(dirs: impl AsRef<str>) -> Vec<Directive> {
-		dirs.as_ref()
+	use sc_tracing::parse_default_directive;
+
+	// Accept all valid directives and print invalid ones
+	fn parse_user_directives(mut env_filter: EnvFilter, dirs: &str) -> EnvFilter {
```

Comment:
I tried out this change of return type, but I reverted it because I think it means:

1. Have to duplicate code in thew 4 other places it's called
2. We bail on first bad directive, don't accept the rest and only print the first bad one (in case there may be more than 1 invalid)

Would it make more sense to rename the `fn` name to `add_directives_and_print_invalid` ?

Reply:
This is completely fine to only bail out the first one. This is about telling the user that he tried an invalid configuration parameter.
He will find out about the others when fixing the first one.

This is very unlikely anyway, however I would like to have this. We already had it twice that we didn't accepted a particular directive, but silently ignored it.

Reply:
yeah, before this PR it was silently ignored. So just to clarify, do you want to just `eprintln` the bad directive, ignore the remaining (possibly valid) directives but still start the node? 

Currently it does actually `eprintln` all of the invalid directives so the user should be aware; and accept all valid ones.

Reply:
I for one would be ok to halt the startup process and exit, with an error message `Invalid logging directive: x` (and if the next is invalid too, rinse&repeat until they get it right): users tend to not read logs very carefully so an `eprintln` will be missed.

Reply:
OK, I got it now. The behavior before this PR was if `init_logger` fails the node continues to start - I'll fix that.

Reply:
Yeah, before it continued. I want, as @dvdplm that we return an error and exit the node. An eprintln would just be ignored.